docs/dev-client-gameplay.md

```diff
@@ -253,475 +454,702 @@ const canBeSpawnPoint = Waypoint.flags[eid] & WaypointFlags.canBeSpawnPoint;
 
-In Hubs, we define the `Text` component as a tag (i.e. with no properties):
+In Hubs, we define the `TextTag` component without any property.
 
 ```typescript
-export const Text = defineComponent();
+export const TextTag = defineComponent();
+// Not
+// export const TextTag = defineComponent({ text: Types.ui32 });
+// TextTag.text[$isStringType] = true;
 ```
 
-We do not duplicate the `text` string in a `bitECS` component. We simply operate on the underlying `Object3D` (a `TroikaText`):
+We do not duplicate the `text` string in a `bitECS` component. We simply
+operate on the underlying `Object3D` (a `TroikaText`):
 
 ```typescript
-const timeLabel = world.eid2obj.get(VideoMenu.timeLabelRef[eid])! as TroikaText;
-timeLabel.text = `${timeFmt(video.currentTime)} / ${timeFmt(video.duration)}`;
-timeLabel.sync();
+const textQuery = defineQuery([TextTag, TimeLabel]);
+export function textSystem(world: HubsWorld) {
+  textQuery(world).forEach(eid => {
+    const timeLabel = world.eid2obj.get(eid)! as TroikaText;
+    timeLabel.text = new Date().toISOString();
+    timeLabel.sync();
+  });
+}
 ```
 
+Refer to the later sections for `world.eid2obj.get(eid)! as XXX`.
 
-<a id="orgbad6d63"></a>
-
-# Adding entities
+TODO: Write an advantage of this pattern.
 
+TODO: Do we use another example because this Text example doesn't match the
+actual implmentation. `TroikaText.sync()` shouldn't be called each system.
+It should be called only once in a frame in a dedicated system. This example
+might be confusing if the readers check the actual code.
 
-<a id="org9d285be"></a>
 
-## Entity basics
+## Writing inflators
 
-Entities are added to the world with `addEntity` and removed from the world with `removeEntity`. In `bitECS`, component state is stored in large, pre-allocated `TypedArrays`. In other words, entities are not objects with components inside them. Entities are simply numbers (`EntityID` s), and the `World` keeps track of things like whether an entity has a given component (`hasComponent(world, MyComponent, eid)`).
+`Inflator` is one of the special concepts, which may not appear in standard
+ECS patterns, in Hubs Client. 
 
-You will rarely need to call `addEntity` yourself. Instead, you will write entity definitions (`EntityDef` s) using `jsx` or create model files (`gltf`) with that add entities and components when the models are loaded.
+Once component is added to an entity and its data is initialized, it starts to
+be processed by systems in ECS world. In other words, components need points
+where they are added to entity and their data is initialized.
 
-Note: We support both `glTF` formats, where binary data buffers contain base64-encoded strings (as in `.gltf`) or raw byte arrays (as in `.glb`). We refer to `gltf` and `glb` files interchangably.
+`Inflator` is the component entry point in the Hubs Client.
 
+TODO: Rename inflate to add or setup because the name may not clearly explain

```

Comment:
I like inflator. The combines combines “setup” and “add” so I think it makes sense to have a separate unique term. This is not a strongly held opinion.

Reply:
Thanks for the comment. Let's privately keep the discussion.